agent/bench-scripts/postprocess/fio-postprocess

```diff
@@ -1,201 +1,292 @@
-while ( <JS> ) {
-	# skip past the non json stuff
-	if ( /{/ ) {
-		$js_str = "{\n";
-		while ( <JS> ) {
-			$js_str = $js_str . $_;
-			if ( /^}/ ) {
-				last;
-			}
+
+
+# %workload
+#		   {Throughput|Latency|Resource|Efficiency}
+#		   {Throughput}[iops_sec, 
+
+my %workload;  # root hash for all data, contains hash refs to
+		# %throughput, %latency, %resource, %efficiency
+my %parameters;	# a hash of parameter-type:array-of-parameter-values
+
+my @benchmark;	# each array element contains a hash with paramters
+		# for this benchmark:
+		# block-size: 256
+		# test-type: randr
+
+my %resource;	# a hash of resource-type:array-of-resource-values,
+
+my @cpu_busy;	# each array element contains a hash with
+		# hostname: hostname or IP
+		# role: client, server, host, kvm-host, container-host, etc.
+		# timeseries: a hash of timestamp:value key-pairs
+my %efficiency;
+my %latency;    # a hash of latency-type:array-of-latency-values,
+		# for example $latency{usec[0..1]
+my @clat;
+my @slat;
+my @lat;	# each array element contains a hash with:
+	    	# hostname: <hostname or IP>
+	    	# role: <client or server>
+	    	# timeseries: a hash of timstamp:value elements
+
+my %throughput; # a hash of throughput-type:array-of-throughput-values,
+		# for example $throughput{MB_sec[0..1]
+my @MB_sec;	# each array element contains a hash with:
+	    	# hostname: <hostname or IP>
+	    	# role: <client or server>
+	    	# timeseries: a hash of timstamp:value key-pairs
+my @iops_sec;	# each array element contains a hash with:
+	    	# hostname: <hostname or IP>
+	    	# role: <client or server>
+	    	# timeseries: a hash of timstamp:value key-pair
+
+my %benchmark_parameters_dataset = ( 	get_label('benchmark_name_label') => $benchmark_name,
+					get_label('benchmark_version_label') => $benchmark_version,
+					#get_label('test_type_label') => $test_type,
+					#get_label('block_size_kbytes_label') => int $block_size_kbytes,
+					#get_label('clients_label') => $clients,
+					get_label('primary_metric_label') => $primary_metric,
+					get_label('uid_label') => create_uid('benchmark_name_label', 'controller_host_label')  );
+
+push(@benchmark, \%benchmark_parameters_dataset);
+
+# Get the tool data
+# Define a set of tool directories which we want to use to report CPU and efficiency metrics
+# Search for tool directories which match the $tool_label_pattern
+my $this_tool_dir;
+my %tool_ids;
+my $tool_group_dir = "$dir/tools-$tool_group";
+my $dh;
+if (opendir($dh, $tool_group_dir)) {
+	foreach $this_tool_dir (sort readdir($dh)) {
+		if ($this_tool_dir =~ /^$tool_label_pattern/) {
+			my $tool_dir_id = $this_tool_dir;
+			$tool_dir_id =~ s/^$tool_label_pattern//;
+			$this_tool_dir = $tool_group_dir . "/" . $this_tool_dir;
+			$tool_ids{$this_tool_dir} = $tool_dir_id;
 		}
 	}
+} else {
+	print "$script: could not find any directories in $tool_group_dir which matched $tool_label_pattern\n";
+}
+closedir  $dh;
```

Comment:
Should the closedir() be in the if branch? I get complaints about invalid dir handles.